src/SourceGenerators/System.Xaml/System.Xaml/ParsedMarkupExtensionInfo.cs

```diff
@@ -49,20 +49,26 @@ public static ParsedMarkupExtensionInfo Parse(string raw, IXamlNamespaceResolver
 
 			if (raw.Length == 0 || raw[0] != '{')
 			{
-				throw Error("Invalid markup extension attribute. It should begin with '{{', but was {0}", raw);
+				throw Error("Invalid markup extension attribute. Expected '{{' not found at the start: \"{0}\"", raw);
 			}
 
 			if (raw.Length >= 2 && raw[1] == '}')
 			{
-				throw Error("Markup extension can not begin with an '{}' escape: '{0}'", raw);
+				throw Error("Markup extension can not begin with an '{}' escape: \"{0}\"", raw);
 			}
 
 			var ret = new ParsedMarkupExtensionInfo();
 			if (raw[raw.Length - 1] != '}')
 			{
 				// Any character after the final closing bracket is not accepted. Therefore, the last character should be '}'.
 				// Ideally, we should still ran the entire markup through the parser to get a more meaningful error.
-				throw Error("Expected '}}' in the markup extension attribute: '{0}'", raw);
+				if (raw.TrimEnd() is string trimmed && trimmed[trimmed.Length - 1] == '}')
```

Comment:
Should we be doing a similar check at the beginning of the expression for any white space?

Reply:
No, because it would be a literal string and not binding anymore.
```xml
<TextBlock Text=" {Binding}" />
```
This would display the text as is.